fuel-types/src/array_types.rs

```diff
@@ -334,7 +334,7 @@ macro_rules! key_methods {
             {
                 use serde::de::Error;
                 if deserializer.is_human_readable() {
-                    let s: &str = serde::Deserialize::deserialize(deserializer)?;
+                    let s: String = serde::Deserialize::deserialize(deserializer)?;
```

Comment:
Deserialization from the file for the `&str` type returns an error because getting a reference to the memory is impossible.